src/uct/ib/base/ib_device.h

```diff
@@ -205,11 +213,28 @@ const uct_ib_device_spec_t* uct_ib_device_spec(uct_ib_device_t *dev);
  * @param port_num              Port number.
  * @param md_config_index       Gid index from the md configuration.
  * @param ib_gid_index          Filled with the selected gid index.
+ * @param roce_ver              Filled with the RoCE version if the link layer
+ *                              is Ethernet.
  */
 ucs_status_t uct_ib_device_select_gid_index(uct_ib_device_t *dev,
                                             uint8_t port_num,
                                             size_t md_config_index,
-                                            uint8_t *ib_gid_index);
+                                            uint8_t *ib_gid_index,
+                                            uct_ib_roce_version_desc_t *roce_ver);
+
+
+/**
+ * Set information for the RoCE port - gid index, RoCE version and address family.
+ *
+ * @param dev                   IB device.
+ * @param port_num              Port number.
+ * @param gid_index             Filled with the selected gid index.
+ * @param roce_ver              Filled with port's RoCE version and address family.
+ */
+ucs_status_t uct_ib_device_set_roce_info(uct_ib_device_t *dev,
```

Comment:
you define `uct_ib_device_roce_info_t` and `uct_ib_roce_version_desc_t ` types, seems like this func should be `uct_ib_device_set_roce_version`. Also pls add [in/out] to @ param descriptions

Reply:
maybe break it to `uct_ib_device_find_best_gid_index` (or `uct_ib_device_find_best_gid_info` to avoid extra call of `uct_ib_device_query_gid_info`) and `uct_ib_device_get_roce_version(dev, gid_idx/gid_info, roce_ver)`?

Reply:
where is the extra call of `uct_ib_device_query_gid_info`?
also, this function gets both the gid (for gid_index selection) and the roce version. so if i split, i would have to to call this function twice. now i call in once from `uct_ib_device_set_roce_info`

Reply:
> where is the extra call of uct_ib_device_query_gid_info?

there is no one :) but will be if split to `uct_ib_device_find_best_gid_index` and `uct_ib_device_get_roce_version` but won't be of `uct_ib_device_find_best_gid_info` + `uct_ib_device_get_roce_version`

>  this function gets both the gid (for gid_index selection) and the roce version

IMO, better when single function does single thing